fix(tracing): Remove incorrect references to response headers (#2485)
getsentry/sentry-docs: src/platforms/javascript/common/performance/index.mdx

## Connecting Backend and Frontend Transactions

To connect backend and frontend transactions into a single choherent trace, Sentry uses a `trace_id` value which is propagated between frontend and backend. Depending on the circumstance, this id is transmitted either in a request header or in an HTML `<meta>` tag. Linking transactions in this way makes it possible for you to navigate between them in the Sentry UI, so you can better understand how the different parts of your system are affecting each other.

### Pageload

> When enabling tracing in both frontend and backend and taking advantage of the automatic frontend instrumentation, you can connect the automatically-generated `pageload` transaction on the frontend with the request transaction that serves the page on the backend. Because JavaScript code running in a browser cannot read the headers of the current page, the `trace_id` must be transmitted in the response itself, specifically in a `<meta>` tag in the `<head>` of the HTML sent from your backend.

Reviewer: This one was there on purpose to explain we use meta instead of response headers -- it obviously cannot be in the request header, since the info comes from the backend.

Explaining why the use of a meta tag is perhaps not crucial, so please either revert this line or remove the justification altogether.

Author: Yeah, I wondered about that one... 

Fixed.

## Connecting Backend and Frontend Transactions

> To connect backend and frontend transactions into a single coherent trace, Sentry uses a `trace_id` value which is propagated between frontend and backend. Depending on the circumstance, this id is transmitted either in a request header or in an HTML `<meta>` tag. Linking transactions in this way makes it possible for you to navigate between them in the Sentry UI, so you can better understand how the different parts of your system are affecting each other.

Reviewer: ```suggestion
To connect transactions from different software components into a single coherent trace, Sentry uses a `trace_id` value which is typically propagated via a request header. Linking transactions in this way makes it possible for you to navigate between them in the Sentry UI, so you can better understand how the different parts of your system are affecting each other.
```

The use of `trace_id` to connect transactions is not limited to what this section was originally about: the specific backend => frontend flow.

The suggestion makes it a more general introduction, leaving the details for the next two subsections.

Author: I like it, but in this case this page lives specifically in the browser (and eventually node) docs, so the front/back idea applies. I'm going to keep this in my back pocket, though, in case we ever try to pull this page into `/common` (which we probably will at some point).

### Navigation and Other XHR Requests

> Once a page is loaded, any requests it makes (and any requests your backend makes as a result) are linked through a request header.

Reviewer: This statement can be misleading. Trace propagation via request headers only happens when there is an "active" transaction / span.

Since we're at it, I'd suggest:

```suggestion
Once a page is loaded, every navigation event triggers a `navigation` transaction, and any Fetch or XHR requests made within this transaction will automatically propagate the trace information.
```

Feel free to tweak it.
